hledger-lib/Hledger/Reports/BalanceReport.hs

```diff
@@ -185,6 +195,28 @@ brNegate (is, tot) = (map brItemNegate is, -tot)
   where
     brItemNegate (a, a', d, amt) = (a, a', d, -amt)
 
+-- | Helper to unify a MixedAmount to a single commodity value.
+unifyMixedAmount :: MixedAmount -> Amount
+unifyMixedAmount mixedAmount = foldl combine (num 0) (amounts mixedAmount)
+  where
+    combine amount result =
+      if isReallyZeroAmount amount
+      then result
+      else if isReallyZeroAmount result
+        then amount
+        else if acommodity amount == acommodity result
+          then amount + result
```

Comment:
Nice PR, thanks!

Does this duplicate [normaliseMixedAmount](https://github.com/simonmichael/hledger/blob/master/hledger-lib/Hledger/Data/Amount.hs#L489) (or "normaliseMixedAmountSquashPricesForDisplay") ?

Reply:
I do not think so: normaliseMixedAmount simplifies an MixedAmount as much as possible, while unifyMixedAmount forces the simplification to a single commodity and bails with an user-visible error message if this is not possible.